tool/tsh/proxy.go

```diff
@@ -186,14 +186,28 @@ func onProxyCommandDB(cf *CLIConf) error {
 		keyFile = profile.KeyPath()
 	}
 
-	lp, err := mkLocalProxy(cf.Context, localProxyOpts{
+	proxyOpts := localProxyOpts{
 		proxyAddr: client.WebProxyAddr,
 		protocols: []alpncommon.Protocol{alpncommon.Protocol(database.Protocol)},
 		listener:  listener,
 		insecure:  cf.InsecureSkipVerify,
 		certFile:  certFile,
 		keyFile:   keyFile,
-	})
+	}
+
+	if database.Protocol == defaults.ProtocolMySQL {
```

Comment:
I would encapsulate this logic into a separate method e.g. `getDatabaseALPNProcotol` or something like this.